2017/proposals/ranasinghe-kanchana.md

```diff
@@ -0,0 +1,156 @@
+
+### June 12th - June 16th
+* legend support for imshow
+
+This is an additional improvement attempted. Since heat-maps do not currently have a direct method for implementation of a legend, this will focus on extending the legend support of other scalarMappables. Also the categorical support for legend here will mostly be for the purpose of serving as a foundation for legend support for other scalarMappables. 
+
+### June 19th - June 23th, **End of Phase 1**
+* testing and debugging of imshow function (heat-map API)
+
+### June 26 - June 30th, **Begin of Phase 2**
+* framework outline for norm functionality extensions
+* norm basic functionality including tests and documentation
+
+With regards to the normalization function, in a Matplotlib sense, it encompasses a function mapping different data values to different colors (or color levels). Currently, the different data values to be mapped from are expected to be numerical. This limitation will be worked around to enable support for categorical data (string data). Also during the course of the project focus will be given to the currently available discrete normalization methods (matplotlib.colors.BoundaryNorm) which handle data in a different sense which may be used as a foundation for this additional unit support. 
+
+### July 3rd - July 7th
+* framework outline for cmap functionality extensions
+* cmap basic functionality including tests and documentation
+
+Currently, the process of using matplotlib.pyplot for heat-maps and other scalarMappables includes the creation of a color-map followed by normalization of all available data to fit this color-map. Color-maps plotting functionality is threefold: sequential, divergent and qualitative. Considering the intrinsic discreteness of categorical data, the extension of the color-maps API will focus on its qualitative plotting ability. However the fundamental behavior of the color-maps API will not require any changes for handling categorical data as our approach simply involves the integration of unit support. Therefore it is possible to build this extended support for categorical data while keeping the public API intact. 
+
+### July 10th - July 14th
+* integration of norm and cmap into heatmaps
+
+Given the functionality extended for categorical data, the final task is entwining all this functionality to imshow to allow the plotting of heat-maps from non-numerical values. This would be handled here. 
+
+### July 17th - July 21th, **End of Phase 2**
+* testing and debugging of norm and cmap APIs
+
+### July 24th - July 28th, **Begin of Phase 3**
+* identifying possible scalarMappables to extend categorical color support
```

Comment:
This work should be done at the `scalerMappable` level so that everything gets this functionality 'for free'.